eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -2166,16 +2188,52 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
     skip.foreach(tx => log.info(s"no need to watch txid=${tx.txid}, it has already been confirmed"))
   }
 
-  def doPublish(localCommitPublished: LocalCommitPublished): Unit = {
+  def doPublish(localCommitPublished: LocalCommitPublished, commitments: Commitments): Unit = {
     import localCommitPublished._
 
-    val publishQueue = List(commitTx) ++ claimMainDelayedOutputTx ++ htlcSuccessTxs ++ htlcTimeoutTxs ++ claimHtlcDelayedTxs
+    val publishQueue = commitments.commitmentFormat match {
+      case Transactions.DefaultCommitmentFormat =>
+        val txs = List(commitTx) ++ claimMainDelayedOutputTx ++ htlcSuccessTxs ++ htlcTimeoutTxs ++ claimHtlcDelayedTxs
+        txs.map(tx => PublishAsap(tx, PublishStrategy.JustPublish))
+      case Transactions.AnchorOutputsCommitmentFormat =>
+        val currentFeerate = commitments.localCommit.spec.feeratePerKw
+        val targetFeerate = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(nodeParams.onChainFeeConf.feeTargets.commitmentBlockTarget)
+        val localFundingPubKey = keyManager.fundingPublicKey(commitments.localParams.fundingKeyPath)
+        val channelKeyPath = keyManager.keyPath(commitments.localParams, commitments.channelVersion)
+        val localPerCommitmentPoint = keyManager.commitmentPoint(channelKeyPath, commitments.localCommit.index)
+        val localHtlcBasepoint = keyManager.htlcPoint(channelKeyPath)
+        // If we have an anchor output available, we will use it to CPFP the commit tx.
+        val publishCommitTx = Transactions.makeClaimAnchorOutputTx(commitTx, localFundingPubKey.publicKey).map(claimAnchorOutputTx => {
```

Comment:
See here: https://github.com/ACINQ/eclair/blob/fa759f1e6b079ae381e5d5f245905d5279aa26d2/eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala#L345

If we don't have any output in the commit tx, we don't have an anchor output because it's completely unnecessary.